functions/consumer/tcp-consumer/src/main/java/org/springframework/cloud/fn/consumer/tcp/TcpConsumerConfiguration.java

```diff
@@ -38,28 +37,28 @@
  *
  * @author Gary Russell
  * @author Christian Tzolov
+ * @author Chris Bono
  */
-@Configuration
+@Configuration(proxyBeanMethods = false)
 @EnableConfigurationProperties({TcpConsumerProperties.class, TcpConnectionFactoryProperties.class})
 public class TcpConsumerConfiguration {
 
-	@Autowired
 	private TcpConsumerProperties properties;
-
-	@Autowired
 	private TcpConnectionFactoryProperties tcpConnectionProperties;
 
-	@Qualifier("tcpSinkConnectionFactory")
-	@Autowired
-	private AbstractConnectionFactory connectionFactory;
+	public TcpConsumerConfiguration(TcpConsumerProperties properties,
```

Comment:
Do we consider these configurations as public API? This change (new non-default constructor) removes the default constructor. If someone was constructing one of these manually for some reason, it would not compile anymore. 

Since this is a minor release this should be acceptable - just wanted to call it out though.

Reply:
I don't think the field injection is too evil, but for consistency it really won't be bad to move autowiring via ctor.
This way when end-user would extend this class, he/she would have access to these injected beans in the overridden ctor.